iosApp/iosApp/view/swiftUI/RankImageView.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  RankImageView.swift
+//  iosApp
+//
+//  Created by Andrew Le on 6/23/24.
+//  Copyright © 2024 orgName. All rights reserved.
+//
+
+import SwiftUI
+import Shared
+
+struct RankImageWithTitle: View {
+    var rank: LadderRank?
+    var text: String
+    var imageSize: CGFloat
+    var onClick: () -> (Void)
+    
+    init(rank: LadderRank?, text: String? = nil, imageSize: CGFloat = 84.0, onClick: @escaping () -> Void = {}) {
+        self.rank = rank
+        self.text = text ?? rank?.nameRes.desc().localized() ?? MR.strings().no_rank.desc().localized()
+        self.imageSize = imageSize
+        self.onClick = onClick
+    }
+    
+    var body: some View {
+        VStack {
```

Comment:
Does this automatically center everything horizontally?

Reply:
The VStack has the default horizontal alignment as center, so yes. That's why I haven't needed to add it in plenty of cases